File: backend/Services/Parsers/WeaverParser.cs
using System.Text.RegularExpressions;
using backend.Models;

namespace backend.Services.Parsers
{
    public class WeaverParser : BasicScoreResultParser
    {
        private ILogger<WeaverParser> _logger;

        public WeaverParser(ILogger<WeaverParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override bool CountWinner => false;
        public override bool Default => false;
        public override string GameName => "Weaver";
        public override string? HelpText => "Alternate entry: \"Weaver X\", where X is your score.";
        // \uD83D\uDFE9 = 🟩
        // \u2B1C\uFE0F = ⬜️
        protected override Regex Parser => new Regex($@"{GameName}([^a-zA-Z]+[a-zA-Z]+\n(?<{ScoreGroup}>[^a-zA-Z]+)| (?<{ScoreGroup}>\d+))", RegexOptions.Singleline);
        protected override string? ExtraContent => null;
        public override string Url => "https://wordwormdormdork.com/";

        protected override DailyResult SetScore(DailyResult dailyResult, Match parserResults)
        {
            if (!parserResults.Groups.ContainsKey(ScoreGroup))
            {
                return dailyResult;
            }

            var scoreLines = parserResults.Groups[ScoreGroup].Value;
            if (!string.IsNullOrWhiteSpace(scoreLines))
            {
                if (int.TryParse(scoreLines, out var score))
                {
                    dailyResult.Score = score;
                }
                else
                {
                    dailyResult.Score = scoreLines.Where(c => c == '\n').Count();
                }
            }

            return dailyResult;
        }
    }
}

File: lambdas/src/parsers/weaver-parser.ts
import { DailyResult } from '../models/daily-result';
import { BasicScoreResultParser } from './base/basic-score-parser';
import { getGroup, hasGroup } from './base/result-parser';

export class WeaverParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Weaver';
  readonly helpText = 'Alternate entry: "Weaver X", where X is your score.';
  protected readonly extraContent = null;
  readonly url = 'https://wordwormdormdork.com/';
  
  // Match either:
  // 1. Grid format: "Weaver ... word\n🟩⬜️🟩⬜️..." (count newlines)
  // 2. Simple format: "Weaver 4" (direct number)
  protected readonly parser = new RegExp(
    `${this.gameName}([^a-zA-Z]+[a-zA-Z]+\\n(?<scoreLines>[^a-zA-Z]+)| (?<scoreNum>\\d+))`,
    's'
  );
  
  protected setScore(dailyResult: DailyResult, match: RegExpMatchArray): DailyResult {
    // Check for direct number format first
    if (hasGroup(match, 'scoreNum')) {
      const scoreNum = getGroup(match, 'scoreNum');
      if (scoreNum) {
        dailyResult.score = parseInt(scoreNum, 10);
        return dailyResult;
      }
    }
    
    // Check for grid format (count newlines)
    if (hasGroup(match, 'scoreLines')) {
      const scoreLines = getGroup(match, 'scoreLines');
      if (scoreLines && scoreLines.trim()) {
        dailyResult.score = (scoreLines.match(/\n/g) || []).length;
      }
    }
    
    return dailyResult;
  }
}
